### PR TITLE
Add ADR about GCloud High Availability

### PR DESCRIPTION
**Pull request by pylipp**
[Rendered file](https://github.com/boxwise/boxtribute/blob/adr/gcloud-high-availability/docs/adr/adr_gcloud-high-availability.md)

### CODE REVIEW

**Comment by pylipp** [comment]
[Discuss GCloud High Availability mode incl. action](https://trello.com/c/BYNLaDQh/1377-discuss-gcloud-high-availability-mode-incl-action)

File: docs/adr/adr_gcloud-high-availability.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+## Decision
+
+Turn off HA.
+
+This is performed by editing the primary database instance in the GCloud console [6].
+
+## Consequences
+
+1. *Data safety* In case of a GCloud outage, data is still available in the back-ups (which can be restored when the primary instance runs again but lost its data). Another possibility is to promote the read-replica in place of the failed primary instance.
+1. *Expenses* We have less monthly costs for using GCloud.
```
Comment (pylipp) [inline docs/adr/adr_gcloud-high-availability.md:55]: @HaGuesto can you state how much turning off HA would save us on average per month?

**Review by aerinsol — APPROVED** [review]

File: docs/adr/adr_gcloud-high-availability.md (hunk 1 above)
Comment (aerinsol) [reply docs/adr/adr_gcloud-high-availability.md:55]: 30 EUR per month currently.

**Comment by aerinsol** [comment]
I wonder at what level of loading or usage should we reenable high availability

File: docs/adr/adr_gcloud-high-availability.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+1. *App availability* Without HA, if the primary instance has an outage, we'll have to react on an alert and manually promote the read-replica. Between the start of the outage and the promotion, the app will not be usable. To moderate this inconvenience we could consider one the following (in either case, it still won't be possible for users to perform actions to update data):
+    - configure the boxtribute app to use the read-replica for any user requests to read data (i.e. GraphQL queries). We had this implemented already in a similar way
+    - configure the boxtribute app to fallback to use the read-replica if connecting to the primary database instance fails.
```
Comment (HaGuesto) [inline docs/adr/adr_gcloud-high-availability.md:58]: This sound good. it should also throw an alert in the case it cannot connect to the primary db

File: docs/adr/adr_gcloud-high-availability.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+A primary instance cannot failover to a read replica, and read replicas are unable to failover in any way during an outage. [3]
+
+Our read replica does not have HA mode enabled itself.
+
+Finally, we have nightly back-ups configured on the primary database instance with point-in-time recovery enabled.
+
+## Decision Drivers
+
+1. **Data safety**: we want user and app data to persist even if a GCloud outage happens
```
Comment (HaGuesto) [inline docs/adr/adr_gcloud-high-availability.md:36]: There are some GDPR requirements in here, too.

> 4.7 Availability and Resilience
The Processor has taken the following measures:
Nightly backup of the database.
Automated and manual testing before deployment to production instance of the Software.
Automatic notifications if an error occurs in the Software’s production instance.
The security measures of Subprocessor for hosting can be found here under section 7 and Appendix 2:
https://cloud.google.com/terms/data-processing-terms

**Pull request merged**